Modules/IO/RAW/include/itkRawImageIO.hxx

```diff
@@ -206,50 +265,54 @@ void RawImageIO< TPixel, VImageDimension >
     const SizeValueType numberOfComponents = this->GetImageSizeInComponents();
-
-#define itkWriteRawBytesAfterSwappingMacro(StrongType, WeakType)        \
-  ( this->GetComponentType() == WeakType )                              \
-    {                                                                   \
-    using InternalByteSwapperType = ByteSwapper< StrongType >;          \
-    const SizeValueType numberOfPixels = numberOfBytes/(sizeof(StrongType)); \
-    if ( m_ByteOrder == LittleEndian )                                  \
-      {                                                                 \
-      StrongType *tempBuffer = new StrongType[numberOfPixels];          \
-      memcpy((char *)tempBuffer, buffer, numberOfBytes);          \
-      InternalByteSwapperType::SwapRangeFromSystemToLittleEndian(       \
-        (StrongType *)tempBuffer, numberOfComponents);                  \
-      file.write((char *)tempBuffer, numberOfBytes);                            \
-      delete[] tempBuffer;                                              \
-      }                                                                 \
-    else if ( m_ByteOrder == BigEndian )                                \
-      {                                                                 \
-      StrongType *tempBuffer = new StrongType[numberOfPixels];          \
-      memcpy((char *)tempBuffer, buffer, numberOfBytes);          \
-      InternalByteSwapperType::SwapRangeFromSystemToBigEndian(          \
-        (StrongType *)tempBuffer, numberOfComponents);                  \
-      file.write((char *)tempBuffer, numberOfBytes);                            \
-      delete[] tempBuffer;                                              \
-      }                                                                 \
-    else                                                                \
-      {                                                                 \
-      file.write(static_cast< const char * >( buffer ), numberOfBytes); \
-      }                                                                 \
-    }
-
     // Swap bytes if necessary
-    if itkWriteRawBytesAfterSwappingMacro(unsigned short, USHORT)
-    else if itkWriteRawBytesAfterSwappingMacro(short, SHORT)
-    else if itkWriteRawBytesAfterSwappingMacro(char, CHAR)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned char, UCHAR)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned int, UINT)
-    else if itkWriteRawBytesAfterSwappingMacro(int, INT)
-    else if itkWriteRawBytesAfterSwappingMacro(long, LONG)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned long, ULONG)
-    else if itkWriteRawBytesAfterSwappingMacro(float, FLOAT)
-    else if itkWriteRawBytesAfterSwappingMacro(double, DOUBLE)
+    if ( this->GetComponentType() == USHORT )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned short >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
```

Comment:
Is this the bracket indentation style that the rules will enforce? If yes, I'm OK. Previous conventions recommended the style that the removed lines are showing.

Reply:
This will be the new style.